### ISSUE TITLE
Kraken2 TheiaCoV: target organism abundance is not correctly captured

### ISSUE THREAD

**Issue by cimendes** [theiagen/public_health_bioinformatics#230]
:bug:

### :pencil: Describe the Issue

The [Kraken2 task for TheiaCoV](https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_kraken2.wdl) currently receives a target organism and reports its abundance as a percentage.

The target organism is inherited from the main TheiaCoV workflow and expects the following options:

`String organism = "sars-cov-2" # options: "sars-cov-2", "HIV", "WNV", "MPXV", "flu"`
https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/theiacov/wf_theiacov_illumina_pe.wdl#L23

Given the way that the target organism percentage capture is coded on Kraken's task, the options provided won't ever be found on the report file as the taxa don't match what is present in the database. This is because a simple grep is performed:
```
    # capture target org percentage 
    if [ ! -z "~{target_org}" ]; then 
      echo "Target org designated: ~{target_org}"
      percent_target_org=$(grep "~{target_org}" ~{samplename}_kraken2_report.txt | cut -f1 | head -n1 )
      if [-z "$percent_target_org" ] ; then percent_target_org="0" ; fi
    else 
      percent_target_org=""
    fi
    echo $percent_target_org | tee PERCENT_TARGET_ORG
```
https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_kraken2.wdl#L36-L44

### :fishing_pole_and_fish: Expected Behavior

It is expected that if, for example, "flu" is selected, the value for "Influenza" abundance is retrieved. Likewise for the other organisms in TheiaCoV. The only exception is sars-cov-2 where the abundance is captured in a separate code block:

https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/taxon_id/task_kraken2.wdl#L28-L33

### :floppy_disk: Version Information

v1.2.1

**Comment by sage-wright** [comment]
Right now, our input JSONs typically include the `target_organism` field, which is what the organism name Kraken2 needs to look for. The `target_organism` field is _not_ inherited from the `organism` field, but is a separate input.

https://github.com/theiagen/public_health_bioinformatics/blob/ab5441926e0a62d60996871639d2f9826ebee052/workflows/utilities/wf_read_QC_trim_pe.wdl#L55

**Comment by cimendes** [comment]
Ops, my bad. I thought from reading the code it was all the same `organism` tag. I would say that remembering the exact term for `target_organism` is too difficult is one is not using the JSON. Shouldn't the target organism be inherited them from the main `organism` tag? It would be a relatively simple adjustment to have an if statement similar to a switch case to check the various options. What do you think?

**Comment by sage-wright** [comment]
Absolutely -- I think that would be an easy way to fix this if people aren't using the JSON files. 

To-do:
- [ ] create an if/else where `target_organism` is determined based on `organism` tag

It might also be a good idea to do this with `vadropts` and other organism-specific parameters that don't change frequently like Docker images

**Comment by sage-wright** [comment]
Resolved via #310
